api/include/opentelemetry/trace/propagation/httptextformat.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <cstdint>
+#include "opentelemetry/context/context.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace trace
+{
+namespace propagation
+{
+
+virtual static Context SetSpanInContext(Span span, Context context = NULL)
+virtual static Span GetCurrentSpan(Context context = NULL)
+
+template <typename T>
+class HTTPTextFormat {
+    /** This class provides an interface that enables extracting and injecting
+    context into headers of HTTP requests. HTTP frameworks and clients
+    can integrate with HTTPTextFormat by providing the object containing the
+     headers, and a getter and setter function for the extraction and
+    injection of values, respectively.*/
```

Comment:
Please put the comment just before the class rather than in it

As a minor point: use the name of the class instead of "This class" (systems like GitHub can pull the comment out and display is when the type is used in other places).

I believe Open Telemetry prefers // style comments over /* */